### PR TITLE
OTP custom sub-modes labeling

### LINKED ISSUE #621: Need to render a different name/content for different transit services that use the same OTP mode.
- For TriMet, "MAX" or "Portland Streetcar" should be shown in batch itinerary results instead of the generic "Tram" or equivalent for the `TRAM` OTP mode.
- Some agencies may want to display "Local bus" or "Express bus" in batch itinerary result headers depending on the dominant bus route (OTP mode: `BUS`).

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Fix #621.
Requires/blocked by #620. 

This PR adds support for a custom `TransitModes` component that renders a mode name based on an OTP itinerary leg.

### CODE REVIEW

**Review by philip-cline — COMMENTED** [review]
Works well for me!

**Review by philip-cline — APPROVED** [review]

File: lib/components/util/formatted-mode.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * JSX wrapper for the i18n.getFormattedMode function (renders into a React fragment).
+ */
+const FormattedMode = ({ mode = '' }: Props): ReactElement => {
+  const intl = useIntl()
+  return <>{getFormattedMode(mode, intl)}</>
+}
+
+export default FormattedMode
```
Comment (daniel-heppner-ibigroup) [inline lib/components/util/formatted-mode.tsx:18]: What if we moved this into the file where getFormattedMode() is? Or are we keeping the components separate?

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM, just wondering if we could consolidate some files?

File: lib/components/util/formatted-mode.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/util/formatted-mode.tsx:18]: Or should we just get rid of this component?

Comment (binh-dam-ibigroup) [reply lib/components/util/formatted-mode.tsx:18]: No, that would add extra lines for adding `useIntl()`. It's better to keep components separate imo.

**Pull request merged**